File: hashing/0387-first-unique-character-in-a-string/java/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int firstUniqChar(String s) {
        Map<Character, Integer> counts = new HashMap<>();

        for (char c : s.toCharArray()) {
            counts.merge(c, 1, Integer::sum);
        }

        for (int i = 0; i < s.length(); i++) {
            if (counts.get(s.charAt(i)) == 1) {
                return i;
            }
        }

        return -1;
    }
}

File: hashing/0387-first-unique-character-in-a-string/cpp/Solution.cpp
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> counts;

        for (char c : s) {
            counts[c]++;
        }

        for (int i = 0; i < (int)s.size(); i++) {
            if (counts[s[i]] == 1) {
                return i;
            }
        }

        return -1;
    }
};
